docs/releases/changelog-dev.md

```diff
@@ -2,8 +2,117 @@
 
 ### New features since last release
 
+- Added `Calibration` class to manage calibrated operations for QProgram, including methods to add (`add_operation`), retrieve (`get_operation`), save (`dump`), and load (`load`) calibration data.
+
+  Example:
+
+  ```Python
+  # Create a Calibration instance
+  calibration = Calibration()
+
+  # Define waveforms
+  drag_wf = IQPair.DRAG(amplitude=1.0, duration=40, num_sigmas=4.5, drag_coefficient=-2.5)
+  readout_wf = ql.IQPair(I=ql.Square(amplitude=1.0, duration=200), Q=ql.Square(amplitude=0.0, duration=200))
+
+  # Add waveforms to the calibration as named operations
+  calibration.add_operation(bus='drive_q0_bus', operation='Xpi', waveform=drag_wf)
+  calibration.add_operation(bus='readout_q0_bus', operation='Measure', waveform=readout_wf)
+
+  # Save the calibration data to a file
+  calibration.dump('calibration_data.yml')
+
+  # Load the calibration data from a file
+  loaded_calibration = Calibration.load('calibration_data.yml')
+  ```
+
+  The contents of `calibration_data.yml` will be:
+
+  ```YAML
```

Comment:
I agree with all the changes from the renaiming from calibrated to WithName, the addition of measure operation (maybe we have to adjust these operations after some requirements from hardware (i.e. Ramiro is concerned on how to handle the wheights and to give the same optios as QM is giving) but it is easy to implement from your structure) and the addition of the YAML structure here.

My only question is why there are some exclamation points on the names and what it is this &id001 (also is this mandatory?) 
(maybe is because I am looking this from GitHub and is changing the format)

Reply:
Instead of saving it like this:

```yaml
drive_q0_bus:
      Xpi: !IQPair
      I: !Gaussian {amplitude: 1.0, duration: 40, num_sigmas: 4.5}
      Q: !DragCorrection
          drag_coefficient: -2.5
          waveform: !Gaussian {amplitude: 1.0, duration: 40, num_sigmas: 4.5}
```

that would create **two different** instances of gaussian waveforms, the library stores it with references so it will create only one when parsing the data. (the same way it is when instantiating with `IQPair.DRAG()`.

Reply:
Also the parent class of all actions in qprogram is already `Operation`.

Reply:
Maybe GroupedOperation for cases like the CNOT ir iSWAP? but well thought about naming confusion and trying to avoid that.